File: HackerRank/C#/ExtraLongFactorials.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;
using System.Numerics;

class Result
{

    /*
     * Complete the 'extraLongFactorials' function below.
     *
     * The function accepts INTEGER n as parameter.
     */
    
    private static BigInteger Factorial(BigInteger n){
        if(n <= 1) return n;
        return n * Factorial(n - 1);
    }
    public static void extraLongFactorials(int n)
    {
        Console.WriteLine(Factorial(n));
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        int n = Convert.ToInt32(Console.ReadLine().Trim());

        Result.extraLongFactorials(n);
    }
}

File: HackerRank/python/extra_long_factorials.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'extraLongFactorials' function below.
#
# The function accepts INTEGER n as parameter.
#

def factorial(n):
    if n <= 1:
        return n;
    return n * factorial(n-1);

def extraLongFactorials(n):
    # Write your code here
    print(factorial(n));

if __name__ == '__main__':
    n = int(input().strip())

    extraLongFactorials(n)
